.github/workflows/CI.yml

```diff
@@ -655,6 +655,12 @@ jobs:
             terraform init
             if terraform apply -auto-approve -lock=false $opts -var="testing_ami=${{ matrix.testing_ami }}" -var="aoc_version=${{ needs.e2etest-preparation.outputs.version }}" -var="testcase=../testcases/${{ matrix.testcase }}" ; then terraform destroy -auto-approve ; else terraform destroy -auto-approve && exit 1 ; fi
```

Comment:
If we are using always, then is there any reason for us to having the ```terraform destroy``` in this code?

Reply:
I guess not for the pass but we do for the fail since we restart this step with the retry action

Reply:
Would it be better to use the retry with [this line](https://github.com/aws-observability/aws-otel-collector/pull/829/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R659-R660)? We can also retry if ```terraform destroy`` failed as well when we cancel the workflow as well?

Reply:
I don't think retry will trigger on cancel.

Reply:
This will now only trigger on cancel